homeassistant/components/sense/sensor.py

```diff
@@ -23,260 +27,247 @@
 from .const import (
     ACTIVE_TYPE,
     CONSUMPTION_ID,
-    CONSUMPTION_NAME,
     FROM_GRID_ID,
-    FROM_GRID_NAME,
     NET_PRODUCTION_ID,
-    NET_PRODUCTION_NAME,
     PRODUCTION_ID,
-    PRODUCTION_NAME,
     PRODUCTION_PCT_ID,
-    PRODUCTION_PCT_NAME,
     SOLAR_POWERED_ID,
-    SOLAR_POWERED_NAME,
     TO_GRID_ID,
-    TO_GRID_NAME,
 )
-from .coordinator import SenseRealtimeCoordinator, SenseTrendCoordinator
+from .coordinator import SenseCoordinator
 from .entity import SenseDeviceEntity, SenseEntity
 
 # Sensor types/ranges
 TRENDS_SENSOR_TYPES = {
-    Scale.DAY: "Daily",
-    Scale.WEEK: "Weekly",
-    Scale.MONTH: "Monthly",
-    Scale.YEAR: "Yearly",
-    Scale.CYCLE: "Bill",
+    Scale.DAY: "daily",
+    Scale.WEEK: "weekly",
+    Scale.MONTH: "monthly",
+    Scale.YEAR: "yearly",
+    Scale.CYCLE: "bill",
 }
 
-# Production/consumption variants
-SENSOR_VARIANTS = [(PRODUCTION_ID, PRODUCTION_NAME), (CONSUMPTION_ID, CONSUMPTION_NAME)]
-
 # Trend production/consumption variants
 TREND_SENSOR_VARIANTS = [
-    *SENSOR_VARIANTS,
-    (PRODUCTION_PCT_ID, PRODUCTION_PCT_NAME),
-    (NET_PRODUCTION_ID, NET_PRODUCTION_NAME),
-    (FROM_GRID_ID, FROM_GRID_NAME),
-    (TO_GRID_ID, TO_GRID_NAME),
-    (SOLAR_POWERED_ID, SOLAR_POWERED_NAME),
+    PRODUCTION_ID,
+    CONSUMPTION_ID,
+    NET_PRODUCTION_ID,
+    FROM_GRID_ID,
+    TO_GRID_ID,
 ]
+TREND_SENSOR_PCT_VARIANTS = [PRODUCTION_PCT_ID, SOLAR_POWERED_ID]
 
 
-async def async_setup_entry(
-    hass: HomeAssistant,
-    config_entry: SenseConfigEntry,
-    async_add_entities: AddEntitiesCallback,
-) -> None:
-    """Set up the Sense sensor."""
-    data = config_entry.runtime_data.data
-    trends_coordinator = config_entry.runtime_data.trends
-    realtime_coordinator = config_entry.runtime_data.rt
+class CoordinatorType(Enum):
```

Comment:
I'd create 2 lists instead of 1 big list where you find out which one belongs where